lib/core/versions/latest/util/Jws.ts

```diff
@@ -128,15 +147,23 @@ export default class Jws {
    * Signs the given payload as a compact JWS string.
    * This is mainly used by tests to create valid test data.
    */
-  public static signAsCompactJws (payload: object, privateKey: any): string {
-    const compactJws = JWS.sign(payload, privateKey);
+  public static signAsCompactJws (payload: object, privateKey: any, protectedHeader?: object): string {
+    const compactJws = JWS.sign(payload, privateKey, protectedHeader);
     return compactJws;
   }
 
   /**
    * Parses the input as a `Jws` object.
+   * @param expectKidInHeader If set to `true`, the given compact JWS must contain `kid`; else the compact JWS must not contain `kid`.
+   */
+  public static parseCompactJws (compactJws: any, expectKidInHeader: boolean): Jws {
+    return new Jws(compactJws, expectKidInHeader);
+  }
+
+  /**
+   * Creates a compact JWS string using the given input. No string validation is performed.
    */
-  public static parse (input: any): Jws {
-    return new Jws(input);
+  public static createCompactJws (protectedHeader: string, payload: string, signature: string): string {
```

Comment:
Nit: looks like this function could be private? Understand if you want to keep it public to be symmetrical with `parseCompactJws`.

Reply:
It's public so tests can use it (to create invalid data mostly).